doc/source/serve/deployment-graph.ipynb

```diff
@@ -0,0 +1,509 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "32ef8771",
+   "metadata": {},
+   "source": [
+    "\n",
+    "## Serve Deployment Graph\n",
+    "\n",
+    "```{note} \n",
+    "Note: This feature is still experimental in Alpha release, some APIs are subject to change.\n",
+    "```\n",
+    "\n",
+    "### General Motivation\n",
+    "\n",
+    "Production machine learning serving pipelines are getting longer and wider. They often consist of multiple, or even tens of models collectively making a final prediction, such as image / video content classification and tagging, fraud detection pipeline with multiple policies and models, multi-stage ranking and recommendation, etc.\n",
+    "\n",
+    "Meanwhile, the size of a model is also growing beyond the memory limit of a single machine due to the exponentially growing number of parameters, such as GPT-3, sparse feature embeddings in recsys models such that the ability to do disaggregated and distributed inference is desirable and future proof.\n",
+    "\n",
+    "We want to leverage the programmable and general purpose distributed computing ability of Ray, double down on its unique strengths (scheduling, communication and shared memory) to facilitate authoring, orchestrating, scaling and deployment of complex serving pipelines under one set of DAG API, so a user can program & test multiple models or multiple shards of a single large model dynamically, deploy to production at scale, and upgrade individually.\n",
+    "\n",
+    "### Key requirements\n",
+    "- Provide the ability to author a DAG of serve nodes to form a complex inference graph.\n",
+    "- Pipeline authoring experience should be fully python programmable with support for dynamic selection, control flows, user business logic, etc.\n",
+    "- DAG can be instantiated and locally executed using tasks and actors API\n",
+    "- DAG can be deployed via declarative and idempotent API, individual nodes can be reconfigured and scaled indepenently.\n",
```

Comment:
I would remove this requirement. `serve run` is not idempotent, and it's only declarative when using the config file, which is undocumented.

Reply:
makes sense, we didn't publicize that part